File: bankeBank/Java/ATMMachineAPP.java
import java.util.Scanner;

public class ATMMachineAPP {

    static String name = "";
    static String pin = "";
    static double balance = 0.0;

    public static void createAccount(String firstName, String lastName, String newPin) {
        name = firstName + " " + lastName;
        pin = newPin;
        balance = 0.0;
        System.out.println("Welcome " + name + "! Your account was created successfully.");
        System.out.println("Your PIN is: " + pin);
        System.out.println("Your starting balance is: " + balance);
    }

    public static void makeDeposit(double depositAmount) {
        if (depositAmount > 0) {
            balance = balance + depositAmount;
            System.out.println("Amount deposited: " + depositAmount);
            System.out.println("Your balance is now: " + balance);
        } else {
            System.out.println("You cannot deposit a negative or zero amount");
        }
    }

    public static void makeWithdrawal(double withdrawalAmount) {
        if (withdrawalAmount <= 0) {
            System.out.println("You cannot withdraw a negative or zero amount");
        } else if (withdrawalAmount > balance) {
            System.out.println("Insufficient funds. Your balance is " + balance);
        } else {
            balance = balance - withdrawalAmount;
            System.out.println("Amount withdrawn: " + withdrawalAmount);
            System.out.println("Your balance is now: " + balance);
        }
    }

    public static void makeTransfer(double transferAmount) {
        if (transferAmount <= 0) {
            System.out.println("You cannot transfer a negative or zero amount");
        } else if (transferAmount > balance) {
            System.out.println("Insufficient funds. Your balance is " + balance);
        } else {
            balance = balance - transferAmount;
            System.out.println("Amount transferred: " + transferAmount);
            System.out.println("Your balance is now: " + balance);
        }
    }

    public static void checkBalance() {
        System.out.println("Your bank balance is: " + balance);
    }

    public static void changePin(String oldPin, String newPin) {
        if (pin.equals(oldPin)) {
            pin = newPin;
            System.out.println("Pin changed successfully");
        } else {
            System.out.println("Incorrect pin");
        }
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        while (true) {
            System.out.println("Welcome to Cazak Global Financial Institude; aka 'money na water' (CGFI)");

            System.out.println("We offer the following services");
            System.out.println("1. Create account");
            System.out.println("2. Make a deposit");
            System.out.println("3. Make withdrawal");
            System.out.println("4. Make transfer");
            System.out.println("5. Check bank balance");
            System.out.println("6. Change pin");
            System.out.println("0. To exit");

	System.out.println("<=======================================================>");

            System.out.println("Choose an option: ");
            String userChoice = scan.nextLine();

            switch (userChoice) {
                case "1":
                    System.out.println("Enter your first name: ");
                    String firstname = scan.nextLine();

                    System.out.println("Enter your last name: ");
                    String lastname = scan.nextLine();

                    System.out.println("Enter your pin: ");
                    String newPin = scan.nextLine();

                    createAccount(firstname, lastname, newPin);
	System.out.println("<=======================================================>");
                    break;
	

                case "2":
                    System.out.println("Enter the amount you want to deposit: ");
                    double deposit = scan.nextDouble();
                    makeDeposit(deposit);
	System.out.println("<=======================================================>");
                    break;

                case "3":
                    System.out.println("Enter the amount you want to withdraw: ");
                    double withdrawal = scan.nextDouble();
                    makeWithdrawal(withdrawal);
	System.out.println("<=======================================================>");
                    break;

                case "4":
                    System.out.println("Enter the amount you want to transfer: ");
                    double transfer = scan.nextDouble();
                    makeTransfer(transfer);
	System.out.println("<=======================================================>");
                    break;
	
                case "5":
                    checkBalance();
	System.out.println("<=======================================================>");
                    break;

                case "6":
                    System.out.println("Enter your old pin: ");
                    String oldPin = scan.nextLine();
                    System.out.println("Enter your new pin: ");
                    String changedPin = scan.nextLine();
                    changePin(oldPin, changedPin);
	System.out.println("<=======================================================>");
                    break;
	
                case "0":
                    System.out.println("Goodbye! Thanks for using our services.");
	System.out.println("<=======================================================>");
                    return;

                default:
                    System.out.println("Invalid option, please select a valid option");
            }
        }
    }
}

File: bankeBank/Javascript/AtmMachineAPP.js
const prompt = require('prompt-sync')();

class AtmMachineAPP {
  constructor(name, balance, pin) {
    this.name = "";
    this.balance = balance;
    this.pin = pin;
  }

  create_account(firstName, lastName, pin) {
    this.name = `${firstName} ${lastName}`;
    this.pin = pin;
    console.log(`Welcome ${this.name}! Your account was created successfully`);
  }

  make_a_deposit(deposit_amount) {
    if (deposit_amount > 0) {
      this.balance += deposit_amount;
      console.log(`The amount deposited is ${deposit_amount} and your balance is ${this.balance}`);
    } else {
      console.log("You cannot deposit a negative or 0 amount");
    }
  }

  make_withdrawal(withdrawal_amount) {
    if (withdrawal_amount <= 0) {
      console.log("You cannot withdraw a negative value");
    } else if (withdrawal_amount > this.balance) {
      console.log(`Insufficient funds. Your balance is ${this.balance}`);
    } else {
      this.balance -= withdrawal_amount;
      console.log(`The amount withdrawn is ${withdrawal_amount} and your balance is ${this.balance}`);
    }
  }

  make_transfer(transfer_amount) {
    if (transfer_amount <= 0) {
      console.log("You cannot transfer a negative value");
    } else if (transfer_amount > this.balance) {
      console.log(`Insufficient funds. Your balance is ${this.balance}`);
    } else {
      this.balance -= transfer_amount;
      console.log(`The amount transferred is ${transfer_amount} and your balance is ${this.balance}`);
    }
  }

  check_balance() {
    console.log(`Your bank balance is: ${this.balance}`);
  }

  change_pin(old_pin, new_pin) {
    if (this.pin === old_pin) {
      this.pin = new_pin;
      console.log("Pin changed successfully");
    } else {
      console.log("Incorrect pin");
    }
  }
}

function main() {
  const account = new AtmMachineAPP("Caleb", 20000, "1443");

  while (true) {
    console.log("\nWelcome to Cazak Global Financial Institude; aka 'money na water' (CGFI)");
    console.log("\nWe offer the following services");
    console.log("Press;");
    console.log("1. Create account");
    console.log("2. Make a deposit");
    console.log("3. Make withdrawal");
    console.log("4. Make transfer");
    console.log("5. Check bank balance");
    console.log("6. Change pin");
    console.log("0. To exit");

    let userChoice = prompt("Choose an option: ");

    if (userChoice === "1") {
      let firstName = prompt("Enter your first name: ");
      let lastName = prompt("Enter your last name: ");
      let pin = prompt("Enter your pin: ");
      account.create_account(firstName, lastName, pin);

    } else if (userChoice === "2") {
      let amount = prompt("Enter the amount you want to deposit: ");
      account.make_a_deposit(amount);

    } else if (userChoice === "3") {
      let withdrawal = prompt("Enter the amount you want to withdraw: ");
      account.make_withdrawal(withdrawal);

    } else if (userChoice === "4") {
      let transfer = prompt("Enter the amount you want to transfer: ");
      account.make_transfer(transfer);

    } else if (userChoice === "5") {
      account.check_balance();

    } else if (userChoice === "6") {
      let oldPin = prompt("Enter your old pin: ");
      let newPin = prompt("Enter your new pin: ");
      account.change_pin(oldPin, newPin);

    } else if (userChoice === "0") {
      console.log("Goodbye! Thanks for using our services.");
      break;

    } else {
      console.log("Invalid option, please select a valid option");
    }
  }
}

main();